resources/js/components/utils/QueryBarrier.tsx

```diff
@@ -0,0 +1,50 @@
+import { CreateQueryResult } from "@tanstack/solid-query";
+import { ImSpinner2 } from "solid-icons/im";
+import { JSX, Match, ParentProps, Switch, mergeProps } from "solid-js";
+
+export interface QueryBarrierProps {
+  /**
+   * Element to show, when query is in error state
+   */
+  errorElement?: JSX.Element;
+  /**
+   * Element to show, when query is in pending state
+   */
+  pendingElement?: JSX.Element;
+  /**
+   * List of queries to handle
+   */
+  queries: CreateQueryResult[];
+}
+
+/**
+ * Default handler for tanstack/solid-query's `createQuery` result
+ *
+ * @todo better looking errorElement
+ */
+export function QueryBarrier(props: ParentProps<QueryBarrierProps>) {
+  const merged = mergeProps(
+    {
+      errorElement: <p>error</p>,
+      pendingElement: (
+        <div class="flex justify-center items-center">
+          <ImSpinner2 size={50} class="animate-spin" />,
+        </div>
+      ),
+    },
+    props
+  );
+
+  const isError = () => merged.queries.some(({ isError }) => isError);
+  const isPending = () =>
```

Comment:
Zawsze dokładnie jeden z tych stanów jest prawdziwy, tak? Jak zadeklarujemy isPending jako !isError&&!isSuccess to będzie to bardziej oczywiste, no i trudniejsze do pomylenia się.